core/thread/src/TThreadExecutor.cxx

```diff
@@ -1,15 +1,32 @@
 #include "ROOT/TThreadExecutor.hxx"
 #include "tbb/tbb.h"
+#include "TROOT.h"
+#include <iostream>
 
 namespace ROOT{
+
+  unsigned TThreadExecutor::fPoolSize = 0;
+
   TThreadExecutor::TThreadExecutor():fInitTBB(new tbb::task_scheduler_init()){
   }
 
   TThreadExecutor::TThreadExecutor(size_t nThreads):fInitTBB(new tbb::task_scheduler_init(nThreads)){
+    if (fPoolSize != 0){
```

Comment:
Mmm...Not sure about the warning there. The scheduler doesn't initialize on declaration but in instantiation. And I just realized I forgot to update the default constructor.

We can't get the current pool size from TBB, only the default one. That's why we keep a copy when enabling implicit MT too.